Various fixes and improvements to documentation (#6)
nrfconnect/sdk-softbank-fota: doc/integration.rst

Overview
========

> SoftBank specifies that your certified product must be able to update its modem firmware, regardless other tasks and operations your main application is running.

Reviewer: ```suggestion
SoftBank specifies that your certified product must be able to update its modem firmware, regardless of the tasks run by your application.
```

The SoftBank FOTA library is an independent library that is intended to run together with your application when included in your project. See the below image.

.. figure:: images/softbank_fota_summary.svg
   :alt: Interaction between the SoftBank FOTA library and the user application

   Interaction between the SoftBank FOTA library and the user application

OS layer
========

The SoftBank FOTA library has an OS abstraction layer.
This abstraction layer makes the softbank library independent of some |NCS| modules and underlying implementation of primitives such as timers, non-volatile storage, and heap allocation.

> For more information, see the :file:`sb_fota_os.h` file or the :ref:`sb_fota_os` section for available APIs. It provides an abstraction of the following modules:

Reviewer: Can't comment on those lines, but we should update the list below, I think none of the NCS modules are right...

Author: Updated the list. It should be correct for the current bitbucket revision.

* ``CONFIG_SB_FOTA`` - Enables the Softbank FOTA library
* ``CONFIG_SB_FOTA_AUTOINIT`` - Initializes the library automatically
* ``CONFIG_SB_FOTA_TLS_SECURITY_TAG`` - Security tag (``sec_tag``) for nRF Cloud TLS connection

> * ``CONFIG_SB_FOTA_JWT_SECURITY_TAG`` - ``sec_tag`` for authentication with the cloud.

Reviewer 2: Add CONFIG_SB_FOTA_CLIENT_ID_PREFIX to this list
